python/lsst/ctrl/bps/wms/htcondor/htcondor_service.py

```diff
@@ -81,6 +82,83 @@ def submit(self, workflow):
             htc_submit_dag(workflow.dag, dict())
             workflow.run_id = workflow.dag.run_id
 
+    def list_submitted_jobs(self, wms_id=None, user=None, require_bps=True, pass_thru=None):
+        """Query WMS for list of submitted WMS workflows/jobs.  This should
+        be a quick lookup function to create list of jobs ids for cancel
+        and other functions.
+
+        Parameters
+        ----------
+        wms_id : `int` or `str`, optional
+            Id that can be used by WMS service to look up workflow/job.
+        user : `str`, optional
+            Limit list to submissions by this particular user.
+        require_bps : `bool`
+            Limit list to submissions via bps.
+        pass_thru : `str`, optional
+            Additional arguments to pass through to the specific WMS service.
+
+        Returns
+        -------
+        job_ids : `list` of `Any`
+            Only job ids to be used by cancel and other functions.  Typically
+            this means top-level jobs (i.e., not children jobs).
+        """
+        _LOG.debug("list_submitted_jobs params: wms_id=%s, user=%s, require_bps=%s, pass_thru=%s",
+                   wms_id, user, require_bps, pass_thru)
+        constraint = ""
+
+        if wms_id is None:
+            if user is not None:
+                constraint = f'(Owner == "{user}")'
+        else:
+            # If wms_id represents path, get numeric id
+            try:
+                cluster_id = int(float(wms_id))
+            except ValueError:
+                wms_path = Path(wms_id)
+                if wms_path.exists():
+                    try:
+                        cluster_id, _ = read_dag_log(wms_id)
+                        cluster_id = int(float(cluster_id))
+                    except StopIteration:
+                        cluster_id = 0
+                else:
+                    cluster_id = 0
+
+            if cluster_id != 0:
+                constraint = f"(DAGManJobId == {int(float(cluster_id))} || ClusterId == " \
+                             f"{int(float(cluster_id))})"
```

Comment:
If we're here `cluster_id` is definitely an integer so the conversions in the f-string are redundant.